src/wxProfilerGUI/profilergui.cpp

```diff
@@ -324,6 +326,67 @@ AttachInfo * ProfilerGUI::AttachToProcess(const std::wstring& processId)
 	return attach_info;
 }
 
+AttachInfo * ProfilerGUI::AttachToMainThread(const std::wstring& processId)
+{
+	DWORD processId_dw;
+	try
+	{
+		processId_dw = std::stoi(processId);
+	}
+	catch (const std::exception&)
+	{
+		throw SleepyException("Not valid process id: " + processId);
+	}

```

Comment:
Please refactor the common code in `AttachToProcess`, `AttachToMostBusyThread`, and `AttachToMostBusyThread` to avoid repetition.

Perhaps an `attachMode` enum parameter?